flakeModule.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, config, system, ... }:
+let
+  inherit (lib) mkOption types;
+
+  nixpkgsOverlaySubmodule = types.mkOptionType {
```

Comment:
Yep, the nixos module will have to be merged with this, to avoid duplication. It is indeed a big change, structurally, but you already did most of the discovering with flake-parts. Good catch on the name. I copied a lot from flake-parts, so will be putting you in the authors of the final squash.